crowbar_framework/app/models/service_object.rb

```diff
@@ -1193,6 +1205,17 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["deployment"][@bc_name]["elements"][elem_remove] ||= []
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              unless databag["deployment"][@bc_name]["elements"][elem_remove].include? old_node
+                databag["deployment"][@bc_name]["elements"][elem_remove] << old_node
+              end
+            end
+
+            databag.save
```

Comment:
@MaximilianMeister re my earlier comment about less saving: I guess my suggestion was to save only once we reach the end of the `element_order.each do |elems|` loop. Right now, you save once per role (in worst case), instead of just once.